### PR TITLE
Adding RestActions support for Get Detector API

### PR DESCRIPTION
**Pull request by saratvemulapalli**
*Issue #195 *

*Description of changes:*
Adding RestActions support for Get Detector API

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/main/java/com/amazon/opendistroforelasticsearch/ad/model/AnomalyDetector.java [hunk 1]
```diff
@@ -153,10 +156,60 @@ public AnomalyDetector(
...
+        uiMetadata = input.readMap();
+        schemaVersion = input.readInt();
+        lastUpdateTime = input.readInstant();
+    }
+
     public XContentBuilder toXContent(XContentBuilder builder) throws IOException {
         return toXContent(builder, ToXContent.EMPTY_PARAMS);
     }
 
+    public void writeTo(StreamOutput output) throws IOException {
```
Comment (ylwu-amzn) [inline src/main/java/com/amazon/opendistroforelasticsearch/ad/model/AnomalyDetector.java:196]: Should we override `Writeable.writeTo` method like `Feature` class?

Comment (saratvemulapalli) [reply src/main/java/com/amazon/opendistroforelasticsearch/ad/model/AnomalyDetector.java:196]: Sure thats a good point. 
Will make changes for both AnomalyDetector and AnomalyDetectorJob

**Review by ohltyler — APPROVED** [review]
LGTM, thanks for adding

**Review by ylwu-amzn — APPROVED** [review]
LGTM. Thanks for the change!

**Pull request merged**